Update Dockerfile.experimental from Dockerfile

yosifkit opened the pull request:
The experimental dockerfile was very outdated; we had been using a custom Dockerfile on the build server for the experimental iso.  Now it can be more clear how it is built.

Fixes #1192

tianon commented on Dockerfile.experimental:
```diff
@@ -1,15 +1,12 @@
+	tar -zxvf /tmp/dockerbin.tgz -C "$ROOTFS/usr/local/bin" --strip-components=1 && \
+	rm /tmp/dockerbin.tgz && \
+	chroot "$ROOTFS" docker -v \
+	sed -ri 's/$/-experimental/' $ROOTFS/etc/version \
```
These two lines are missing `&&` in between, so it's probably worth converting this whole file to the "postfix-`&&`" format to ensure we don't miss any (since this file is relatively short, unlike the main `Dockerfile` :sweat:).

yosifkit replied:
:open_mouth: :disappointed:  This time I will build test my changes first.

yosifkit commented:
Updated!

```console
docker build -f Dockerfile.experimental .
Sending build context to Docker daemon 1.674 MB
Step 1 : FROM boot2docker/boot2docker
 ---> 47c3eb7a79c6
Step 2 : RUN curl -fSL -o /tmp/dockerbin.tgz https://experimental.docker.com/builds/Linux/x86_64/docker-$(cat $ROOTFS/etc/version).tgz 	&& tar -zxvf /tmp/dockerbin.tgz -C "$ROOTFS/usr/local/bin" --strip-components=1 	&& rm /tmp/dockerbin.tgz 	&& chroot "$ROOTFS" docker -v 	&& sed -ri 's/$/-experimental/' $ROOTFS/etc/version 	&& cp -v "$ROOTFS/etc/version" /tmp/iso/version
 ---> Running in 0a61eb01aac9
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 27.6M  100 27.6M    0     0  3156k      0  0:00:08  0:00:08 --:--:-- 3705k
docker/docker-containerd-ctr
docker/docker
docker/docker-containerd
docker/dockerd
docker/docker-proxy
docker/docker-runc
docker/docker-containerd-shim
Docker version 1.12.0, build 8eab29e, experimental
'/rootfs/etc/version' -> '/tmp/iso/version'
 ---> 8e7ea8edf48f
Removing intermediate container 0a61eb01aac9
Step 3 : RUN { echo; echo "  WARNING: this is a build from experimental.docker.com, not a stable release."; echo; } >> "$ROOTFS/etc/motd"
 ---> Running in ecc959ea1fcb
 ---> e62e7b087181
Removing intermediate container ecc959ea1fcb
Step 4 : RUN /make_iso.sh
 ---> Running in 615913a5c246
+ xz -9 --format=lzma --verbose --verbose
+ cpio -o -H newc
xz: Filter chain: --lzma1=dict=64MiB,lc=3,lp=0,pb=2,mode=normal,nice=64,mf=bt4,depth=0
xz: 674 MiB of memory is required. The limiter is disabled.
xz: Decompression will need 65 MiB of memory.
380937 blocks
(stdin): 34.0 MiB / 186.0 MiB = 0.183, 2.1 MiB/s, 1:29
/
xorriso 1.3.2 : RockRidge filesystem manipulator, libburnia project.

Drive current: -outdev 'stdio:/boot2docker.iso'
Media current: stdio file, overwriteable
Media status : is blank
Media summary: 0 sessions, 0 data blocks, 0 data, 98.2g free
xorriso : WARNING : -volid text is too long for Joliet (32 > 16)
xorriso : WARNING : -volid text does not comply to ISO 9660 / ECMA 119 rules
Added to ISO image: directory '/'='/tmp/iso'
xorriso : UPDATE : 12 files added in 1 seconds
xorriso : UPDATE : 12 files added in 1 seconds
xorriso : NOTE : Copying to System Area: 432 bytes from file '/usr/lib/ISOLINUX/isohdpfx.bin'
libisofs: NOTE : Aligned image size to cylinder size by 54 blocks
xorriso : UPDATE :  84.21% done
ISO image produced: 19456 sectors
Written to medium : 19456 sectors at LBA 0
Writing to 'stdio:/boot2docker.iso' completed successfully.

 ---> d15e1aa8cf30
Removing intermediate container 615913a5c246
Successfully built d15e1aa8cf30
```

tianon commented:
LGTM :heart:
